crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -332,6 +341,196 @@ impl<P: FsProvider> ResctrlPlugin<P> {
+        let new_state = if res.missing == 0 {
+            ContainerSyncState::Reconciled
+        } else {
+            ContainerSyncState::Partial
+        };
+
+        // Re-acquire lock and update counters/state conditionally
+        let mut st = self.state.lock().unwrap();
+        // Track whether we improved state
+        let mut did_reconcile = false;
+        let ret_state = if let Some(cst) = st.containers.get_mut(container_id) {
+            if cst.state == ContainerSyncState::Partial
+                && new_state == ContainerSyncState::Reconciled
+            {
+                cst.state = ContainerSyncState::Reconciled;
+                did_reconcile = true;
+            }
+            cst.state
+        } else {
+            return Err(resctrl::Error::Io {
+                path: std::path::PathBuf::from("<container>"),
+                source: std::io::Error::from_raw_os_error(libc::ENOENT),
+            });
+        };
+
+        if did_reconcile {
+            if let Some(ps) = st.pods.get_mut(&pod_uid) {
+                ps.reconciled_containers += 1;
+                let snapshot = ps.clone();
+                drop(st);
+                self.emit_pod_add_or_update(&pod_uid, &snapshot);
+                return Ok(ContainerSyncState::Reconciled);
+            }
+        }
+        Ok(ret_state)
+    }
+
+    /// Retry once across all pods/containers.
+    /// Stops group-creation retries on first Capacity error in this pass.
+    pub fn retry_all_once(&self) -> resctrl::Result<()> {
+        // Snapshot lists under lock
+        let (failed_pods, partial_containers): (Vec<String>, Vec<String>) = {
+            let st = self.state.lock().unwrap();
+            let pods = st
+                .pods
+                .iter()
+                .filter_map(|(uid, ps)| {
+                    if matches!(ps.group_state, ResctrlGroupState::Failed) {
+                        Some(uid.clone())
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            let containers = st
+                .containers
+                .iter()
+                .filter_map(|(cid, cs)| {
+                    if cs.state == ContainerSyncState::Partial {
+                        Some(cid.clone())
+                    } else {
+                        None
+                    }
+                })
+                .collect();
+            (pods, containers)
+        };
+
+        // Retry group creation until first capacity error
+        for uid in failed_pods {
+            match self.retry_group_creation(&uid) {
+                Err(resctrl::Error::Capacity { .. }) => break,
+                Err(e) => return Err(e),
+                Ok(_) => {}
+            }
+        }
+
+        // Retry container reconcile for partial containers
+        for cid in partial_containers {
+            let _ = self.retry_container_reconcile(&cid)?;
```

Comment:
ContainerNotFound and PodNotFound errors would means a race with other components of the system, which is benign and should not stop reconciliation